### PR TITLE
Better Bayestree wrapping

### PR DESCRIPTION
**Pull request by dellaert**
Add access to cliques, some more methods.

Add dot file, which renders as (use VS code extension!):

image

### CODE REVIEW

File: gtsam/discrete/discrete.i [hunk 1]
```diff
@@ -92,12 +92,28 @@ class DiscreteBayesNet {
...
 
 #include <gtsam/discrete/DiscreteBayesTree.h>
+class DiscreteBayesTreeClique {
+  DiscreteBayesTreeClique();
+  DiscreteBayesTreeClique(const gtsam::DiscreteConditional* conditional);
+  const gtsam::DiscreteConditional* conditional() const;
+  bool isRoot() const;
+  void printSignature(
+      const string& s = "Clique: ",
+      const gtsam::KeyFormatter& formatter = gtsam::DefaultKeyFormatter) const;
```
Comment (ProfFan) [inline gtsam/discrete/discrete.i:102]: newline before bracket

File: python/gtsam/tests/test_DiscreteBayesTree.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        ordering = Ordering()
+        for j in range(15):
+            ordering.push_back(j)
+        bayesTree = factorGraph.eliminateMultifrontal(ordering)
+
+        # Uncomment these for visualization:
+        # print(bayesTree)
+        # for key in range(15):
+        #     bayesTree[key].printSignature()
+        # bayesTree.saveGraph("test_DiscreteBayesTree.dot")
```
Comment (ProfFan) [inline python/gtsam/tests/test_DiscreteBayesTree.py:76]: Maybe add assertion of equality to the `dot` file contents

**Review by ProfFan — APPROVED** [review]
Looks good sans comments

File: gtsam/discrete/discrete.i (hunk 1 above)
Comment (dellaert) [reply gtsam/discrete/discrete.i:102]: actually, tried re-formatting, stays the same.

**Pull request merged**